srv/components/misc_pkgs/swupgrade/repo/init.sls

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+{% from './install.sls' import repo_added with context %}
+{% from './teardown.sls' import repo_absent with context %}
```

Comment:
I don't think that old swupdate logic is good to base on. Upgrade repo would be lmore like a single repo that we havae for deployment so better to check 
`cortx-prvsnr/api/python/provisioner/srv/salt/repos/`.

Reply:
I adapted this swupdate logic for swupgrade too. Lets consider moving to `cortx-prvsnr/api/python/provisioner/srv/salt/repos/` as new separate ticket

Reply:
Corresponding ticket:
https://jts.seagate.com/browse/EOS-16932